### ISSUE TITLE
[Regression] The non-abstract class 'InputConnectionController' is missing implementations for these members:  - TextInputClient.connectionClosed

### ISSUE THREAD

**Issue by JCKodel** [memspace/zefyr#210]
IMPORTANT: I'm forced to use Flutter's beta channel because of Adreno 308 bug (stable channel leads to a black screen in 15% or our users database).

## Steps to Reproduce

I've upgraded Flutter because VSCode told me to do so.

## Logs

```
Compiler message:
/C:/Flutter/.pub-cache/hosted/pub.dartlang.org/zefyr-0.8.0/lib/src/widgets/input.dart:11:7: Error: The non-abstract class 'InputConnectionController' is missing implementations for these members:
 - TextInputClient.connectionClosed
Try to either
 - provide an implementation,
 - inherit an implementation from a superclass or mixin,
 - mark the class as abstract, or
 - provide a 'noSuchMethod' implementation.

class InputConnectionController implements TextInputClient {
      ^^^^^^^^^^^^^^^^^^^^^^^^^
/C:/Flutter/packages/flutter/lib/src/services/text_input.dart:658:8: Context: 'TextInputClient.connectionClosed' is defined here.
  void connectionClosed();
       ^^^^^^^^^^^^^^^^
Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
#0      KernelSnapshot.build  (package:flutter_tools/src/build_system/targets/dart.dart:213:7)
<asynchronous suspension>
#1      _BuildInstance._invokeInternal  (package:flutter_tools/src/build_system/build_system.dart:526:25)
<asynchronous suspension>
#2      _BuildInstance.invokeTarget.<anonymous closure>  (package:flutter_tools/src/build_system/build_system.dart:481:35)
#3      new Future.sync  (dart:async/future.dart:222:31)
#4      AsyncMemoizer.runOnce 
package:async/src/async_memoizer.dart:43
#5      _BuildInstance.invokeTarget  (package:flutter_tools/src/build_system/build_system.dart:481:21)
<asynchronous suspension>
<asynchronous suspension>
#6      BuildSystem.build  (package:flutter_tools/src/build_system/build_system.dart:419:36)
<asynchronous suspension>
#7      AssembleCommand.runCommand  (package:flutter_tools/src/commands/assemble.dart:154:50)
<asynchronous suspension>
#8      FlutterCommand.verifyThenRunCommand  (package:flutter_tools/src/runner/flutter_command.dart:556:18)
<asynchronous suspension>
#9      FlutterCommand.run.<anonymous closure>  (package:flutter_tools/src/runner/flutter_command.dart:456:33)
<asynchronous suspension>
#10     AppContext.run.<anonymous closure>  (package:flutter_tools/src/base/context.dart:146:29)
<asynchronous suspension>
#11     _rootRun  (dart:async/zone.dart:1124:13)
#12     _CustomZone.run  (dart:async/zone.dart:1021:19)
#13     _runZoned  (dart:async/zone.dart:1516:10)
#14     runZoned  (dart:async/zone.dart:1463:12)
#15     AppContext.run  (package:flutter_tools/src/base/context.dart:145:18)
<asynchronous suspension>
#16     FlutterCommand.run  (package:flutter_tools/src/runner/flutter_command.dart:446:20)
#17     CommandRunner.runCommand 
package:args/command_runner.dart:197
<asynchronous suspension>
#18     FlutterCommandRunner.runCommand.<anonymous closure>  (package:flutter_tools/src/runner/flutter_command_runner.dart:416:21)
<asynchronous suspension>
#19     AppContext.run.<anonymous closure>  (package:flutter_tools/src/base/context.dart:146:29)
<asynchronous suspension>
#20     _rootRun  (dart:async/zone.dart:1124:13)
#21     _CustomZone.run  (dart:async/zone.dart:1021:19)
#22     _runZoned  (dart:async/zone.dart:1516:10)
#23     runZoned  (dart:async/zone.dart:1463:12)
#24     AppContext.run  (package:flutter_tools/src/base/context.dart:145:18)
<asynchronous suspension>
#25     FlutterCommandRunner.runCommand  (package:flutter_tools/src/runner/flutter_command_runner.dart:367:19)
<asynchronous suspension>
#26     CommandRunner.run.<anonymous closure> 
package:args/command_runner.dart:112
#27     new Future.sync  (dart:async/future.dart:222:31)
#28     CommandRunner.run 
package:args/command_runner.dart:112
#29     FlutterCommandRunner.run  (package:flutter_tools/src/runner/flutter_command_runner.dart:251:18)
#30     run.<anonymous closure>.<anonymous closure>  (package:flutter_tools/runner.dart:63:22)
<asynchronous suspension>
#31     _rootRun  (dart:async/zone.dart:1124:13)
#32     _CustomZone.run  (dart:async/zone.dart:1021:19)
#33     _runZoned  (dart:async/zone.dart:1516:10)
#34     runZoned  (dart:async/zone.dart:1500:12)
#35     run.<anonymous closure>  (package:flutter_tools/runner.dart:61:18)
<asynchronous suspension>
#36     AppContext.run.<anonymous closure>  (package:flutter_tools/src/base/context.dart:146:29)
<asynchronous suspension>

#37     _rootRun  (dart:async/zone.dart:1124:13)
#38     _CustomZone.run  (dart:async/zone.dart:1021:19)
#39     _runZoned  (dart:async/zone.dart:1516:10)
#40     runZoned  (dart:async/zone.dart:1463:12)
#41     AppContext.run  (package:flutter_tools/src/base/context.dart:145:18)
<asynchronous suspension>
#42     runInContext  (package:flutter_tools/src/context_runner.dart:65:24)
<asynchronous suspension>
#43     run  (package:flutter_tools/runner.dart:50:10)
#44     main  (package:flutter_tools/executable.dart:65:9)
<asynchronous suspension>
#45     main 
C:\…\bin\flutter_tools.dart:8
#46     _startIsolate.<anonymous closure>  (dart:isolate-patch/isolate_patch.dart:303:32)
#47     _RawReceivePortImpl._handleMessage  (dart:isolate-patch/isolate_patch.dart:172:12)

build failed.

FAILURE: Build failed with an exception.
```

```
[√] Flutter (Channel beta, v1.11.0, on Microsoft Windows [Version 10.0.18362.476], locale pt-BR)
    • Flutter version 1.11.0 at C:\Flutter
    • Framework revision 856a90e67c (2 weeks ago), 2019-11-08 18:00:01 -0800
    • Engine revision af04338413
    • Dart version 2.7.0

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\jckod\AppData\Local\Android\Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_SDK_ROOT = C:\Users\jckod\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 41.0.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] VS Code, 64-bit edition (version 1.40.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.6.0

[√] Connected device (1 available)
    • XT1033 • 0423490246 • android-arm • Android 5.0.2 (API 21)

• No issues found!
```

**Comment by SherwinJames** [comment]
I have the exact same issue.  From the Zefyr changelog, it looks as though flutter version 1.9.0 is officially support.  Wondering what the latest flutter version is actually supported.

**Comment by cc123** [comment]
I have just found the same issue, but only impacting me on Mac.

**Comment by cgestes** [comment]
Please use this while it gets merged to master and released: https://github.com/memspace/zefyr/tree/fix-text-input-connection-closed

**Comment by bpasha7** [comment]
I have the same issue. Please fix it. into your _input.dart_ file you have not implemention for **connectionClosed**() function.  🤞

**Comment by cgestes** [comment]
It just needs to be merged. in the meantime you can use the following in your pubspec:

```yaml
  zefyr:
      git:
        url: git://github.com/memspace/zefyr.git
        ref: fix-text-input-connection-closed
        path: packages/zefyr
```

**Comment by AlexMainaMunyua** [comment]
How can we merge it ??

**Comment by AlexMainaMunyua** [comment]
Though it is working well when i use that extension configuration in my pubspec.yaml

**Comment by cc123** [comment]
Apologies for my ignorance, but where does this get placed in pubspec.yaml? What should I do (if anything) with my current zefyr: ^0.8.0 reference?

**Comment by JEuler** [comment]
@cc123 Just insteaad of `zefyr: ^0.8.0.`

**Comment by pulyaevskiy** [comment]
Thanks @cgestes for helping out!
I just published zefyr 0.9 which fixes support of Flutter 1.12. Sorry everyone for the delay.
